docs/user-guide/timeouts/timeouts-main.md

```diff
@@ -0,0 +1,331 @@
+candidate causes for timeouts, and show ways to sometimes prevent them. We give
+a glimpse into the more theoretical background of timeouts in program
+verification in [this section](timeouts-theory.md)
+
+We classify Certora Prover timeouts as follows:
+1.  Timeouts that happen before SMT solvers are running 
+2.  Timeouts where the SMT queries in sum lead to a global timeout
+3.  Timeouts where a single SMT query could not be solved 
+
+Types 1. and 2. are signified by a hard stop of the Prover. That means the
+Prover ran until the global timeout (set via `--globalTimeout`, typically 2
+hours) and was forcefully shut down from everything it was doing. A message like
+"hard stop reached" appears in the "Global problems" pane of the report, and
+error symbols next to one or many rules.
+
+% made CERT-3797 so we get a clear indication to the user, hopefully
+% then, CVT should indicate via report-logs
+%  - whether there was a hard stop
+%  - whether SMT has started
+
+
+Type 3. is signified by a soft stop. This means an SMT solver shut down due to
+hitting the limit for a single SMT run (set via `--smt_timeout`). When running
+with default settings this means that we give up for the individual rule since
+in order to obtain a proof of correctness we need to solve every subproblem we
+generate (see documentation on [control-flow splitting](control-flow-splitting)
+for more details). 
+
+In the remainder we will focus on the mitigation of SMT timeouts, i.e., types 2.
+and 3. Non-SMT Timeouts (Type 1.) should be reported to Certora. 
+
+(timeout_causes)=
+# What Causes Timeouts?
+
+As a first step towards resolving a timeout, we need to diagnose its root
+causes. In our experience so far, the following are some of the most common
+reasons for SMT timeouts:
+
+ - Non-trivial amount of nonlinear arithmetic
+ - Very high path count
+ - High storage/memory complexity
+
+This list is not exhaustive, but the majority of timeouts we have observed so
+far can be traced back to one or more of these causes. While these are not the
+only sources of complexity, they should work well as general estimates of
+complexity. For instance, linear arithmetic usually only becomes a problem when
+the input program is rather large, which is also indicated by path count in most
+practical cases.
+
+
+## Complexity Feedback from Certora Prover
+
+Certora Prover provides some help with diagnosing timeouts. We present these
+features in this section.
+
+### Difficulty Statistics
+
+Certora Prover provides statistics on the problem sizes it encounters. 
+These statistics are structured according to the timeout reasons given above.
+
+Currently, the Prover tracks the following statistics:
+ - Path count
+ - Nonlinear operations count
+ - Memory/storage complexity
+
+For a very short summary we give one summarizing number for each of the
+statistics, along with a LOW/MEDIUM/HIGH statement. This occurs as an INFO
+message in the Global Problems pane of the Prover reports.
```

Comment:
Was this already merged? I mean the info in the global pane

Reply:
myeah .. no .. might not end up like that after all; commenting for now; will have to update once we actually display these